gbz80-configuration.json

```diff
@@ -17,7 +17,6 @@
     ["\"", "\""]
   ],
   "indentationRules" : {
-    "increaseIndentPattern" : "^(((([a-zA-Z_][a-zA-Z_0-9]*)?\\.)?[a-zA-Z_][a-zA-Z_0-9]*[:]{1,2})|(\\.[a-zA-Z_][a-zA-Z_0-9]*)).*$",
-    "decreaseIndentPattern" : "^\\s+(((([a-zA-Z_][a-zA-Z_0-9]*)?\\.)?[a-zA-Z_][a-zA-Z_0-9]*[:]{1,2})|(\\.[a-zA-Z_][a-zA-Z_0-9]*)).*$"
+    "increaseIndentPattern" : "^(((([a-zA-Z_][a-zA-Z_0-9]*)?\\.)?[a-zA-Z_][a-zA-Z_0-9]*[:]{1,2})|(\\.[a-zA-Z_][a-zA-Z_0-9]*)).*$"
```

Comment:
I'm not as sure about this change.

I see where you're coming from about this rule interfering with your preferred indentation style, and I agree that the extension should better accommodate your style. Making it optional would be preferable to removing it entirely, however. I would like to investigate other possible options, to see if there's a way it could be made configurable. Could you remove this change from this pull request, and open it in its own, so that I could weigh it independently of the other changes?

Reply:
I agree that it would be better if it was configurable, as it works well for a traditional assembly codestyle. I will open it in its own PR once I have time.

Thank you for the response!

Reply:
There are a few alternatives to removing this rule that I've found.

Most significantly, it's a VSCode setting. VSCode has an "Auto Indent" option. You could change it to disable auto indentation. If you wanted to disable auto indents for only this language, you could do so via settings.json.

```json
{
  "[gbz80]": {
    "editor.autoIndent": "keep",
  }
}
```

The option "keep", shown here, has newlines preserve the indentation level of the previous line, ignoring the language-defined indentation rules. There are other options you could also choose.

The JSON language configuration also supports "onEnterRules", though they look like they're just a richer version of "indentationRules", and so probably aren't a way to actually changing the behavior.

Finally, there's VSCode extension formatters, which this extension implements in "formatter.ts". They allow open-ended formatting logic, including the ability to read custom settings. We could add custom, richer formatting rules, which the extension could implement using custom formatter logic.

Of the options available, I think disabling auto-indent on gbz80 in your settings will likely cover your use case, but let me know if that falls short of what would be best for you.

Reply:
I did not know that was an option, that will definitely do for now, thank you!
I have removed the changes to auto-decrementation from the pull request.

I would like to eventually discuss how to improve it. But as of right now, there are other issues I would rather work on.